src/Animal.java

```diff
@@ -0,0 +1,39 @@
+
+public abstract class Animal
+{
+    private static int totalAnimals;
+
+    protected String name;
+    protected int maxRunningDistance;
+    protected int maxSwimmingDistance;
+
+    public Animal(String name, int maxRunningDistance, int maxSwimmingDistance)
+    {
+        this.name = name;
+        this.maxRunningDistance = maxRunningDistance;
+        this.maxSwimmingDistance = maxSwimmingDistance;
+        totalAnimals++;
+    }
+
+
+    //бег
+    void run(int distance)
+    {
+        if (distance > maxRunningDistance)
+        {
+            System.out.printf("%s пробежал(а) %s м. из %s м., т.к. указанное расстояние превышает максимальное расстояние бега.\n", name, maxRunningDistance, distance);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.